Lib/fontTools/ttLib/tables/_c_m_a_p.py

```diff
@@ -38,6 +38,20 @@ def getcmap(self, platformID, platEncID):
 				return subtable
 		return None # not found
 
+	def getBestCmap(self, cmapPreferences=((3, 10), (3, 1), (0, 3))):
+		"""Return the 'best' unicode cmap dictionary available in the font.
+
+		By default it will search for the following (platformID, platEncID)
+		pairs:
+			(3, 10), (3, 1), (0, 3)
+		But this can be customized via the cmapPreferences argument.
+		"""
+		for platformID, platEncID in cmapPreferences:
+			cmapSubtable = self.getcmap(platformID, platEncID)
+			if cmapSubtable is not None:
+				return cmapSubtable.cmap
+		raise ValueError("None of the requested cmap subtables were found")
```

Comment:
why raise instead of simply returning None? Perhaps is better to leave it to the caller to decide what to do in case the requested cmap is missing.

Reply:
Hm, I think I prefer an exception, because that gives the better error message if the caller _doesn't_ check the result. And there's always try/except :)

Reply:
I don't want to insist too much on this, if you prefer the exception. But I think that a ValueError may be too heavy handed. This method is more about "finding" a best cmap than "getting", because you don't know in advance whether there will be one. That's why I thought a None result is more meaningful; it's obvious that it means "I coudn't find any best cmap" and your exception message does not add any more information than "None".
And if one then doesn't check for None and attempts to call cmap methods on NoneType, well Python will surely raise a TypeError, which speaks by itself. And the docstring can make it clear that the result may be None. Anyway, up to you.

Reply:
Hm, I see your point, let me think. (So far, it seems very exceptional if no such cmap is available, hence the exception. But yeah, if a fallback is needed, None is easier to detect.)